service/controller/resource/masters/create_block_api_calls.go

```diff
@@ -0,0 +1,174 @@
+package masters
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-11-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/service/controller/internal/state"
+	"github.com/giantswarm/azure-operator/service/controller/key"
+)
+
+const (
+	temporarySecurityRuleName = "temporaryFlatcarMigration"
+)
+
+func (r *Resource) blockAPICallsTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return currentState, microerror.Mask(err)
+	}
+
+	workersRule := network.SecurityRule{
+		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+			Access:                   network.SecurityRuleAccessDeny,
+			Description:              to.StringPtr("Temporarily block API access from workers during flatcar migration"),
+			DestinationPortRange:     to.StringPtr("443"),
+			DestinationAddressPrefix: to.StringPtr("*"),
+			Direction:                network.SecurityRuleDirectionOutbound,
+			Protocol:                 network.SecurityRuleProtocolAsterisk,
+			Priority:                 to.Int32Ptr(3000),
+			SourceAddressPrefix:      to.StringPtr("*"),
+			SourcePortRange:          to.StringPtr("*"),
+		},
+	}
+
+	mastersRule := network.SecurityRule{
+		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+			Access:                   network.SecurityRuleAccessDeny,
+			Description:              to.StringPtr("Temporarily block internet access to masters during flatcar migration"),
+			DestinationPortRange:     to.StringPtr("*"),
+			DestinationAddressPrefix: to.StringPtr("*"),
+			Direction:                network.SecurityRuleDirectionOutbound,
+			Protocol:                 network.SecurityRuleProtocolAsterisk,
+			Priority:                 to.Int32Ptr(3000),
+			SourceAddressPrefix:      to.StringPtr("*"),
+			SourcePortRange:          to.StringPtr("*"),
+		},
+	}
+
+	workerFound, err := r.ensureSecurityRule(ctx, key.ResourceGroupName(cr), key.WorkerSecurityGroupName(cr), temporarySecurityRuleName, workersRule)
+	if err != nil {
+		return currentState, microerror.Mask(err)
```

Comment:
nitpick :) according to another comment from @tuommaki, we should return `""` here (and in few more other places). See https://github.com/giantswarm/fmt/blob/master/go/errors.md#zero-values-when-returning-errors.
```suggestion
		return "", microerror.Mask(err)
```

Reply:
done